# Análisis Exploratorio de Datos (EDA) - Parte 1: Conocer los datos.

Iniciamos el proyecto haciendo un análisis exploratorio de datos (EDA), nos enfocamos en conocer a fondo los datos del Titanic. Este paso inicial es sumamente necesario para comprender la estructura y características de los datos con los que vamos a trabajar en las siguientes etapas del proyecto.

El objetivo principal de este archivo es revisar las variables, su tipo de datos, la cantidad de registros y verificar si los datasets tienen valores nulos o datos atípicos.

In [1]:
import pandas as pd
import numpy as np

In [2]:
# Cargamos los datos.
train_data = pd.read_csv(r'C:\Users\rodri\Desktop\Titanic 2.0\datos\train.csv')
test_data = pd.read_csv(r'C:\Users\rodri\Desktop\Titanic 2.0\datos\test.csv')

In [3]:
print('Datos de entrenamiento:')
train_data.head()

Datos de entrenamiento:


PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [4]:
print('Datos de prueba:')
test_data.head()

Datos de prueba:


PassengerId  Pclass                                          Name     Sex  \
0          892       3                              Kelly, Mr. James    male   
1          893       3              Wilkes, Mrs. James (Ellen Needs)  female   
2          894       2                     Myles, Mr. Thomas Francis    male   
3          895       3                              Wirz, Mr. Albert    male   
4          896       3  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female   

    Age  SibSp  Parch   Ticket     Fare Cabin Embarked  
0  34.5      0      0   330911   7.8292   NaN        Q  
1  47.0      1      0   363272   7.0000   NaN        S  
2  62.0      0      0   240276   9.6875   NaN        Q  
3  27.0      0      0   315154   8.6625   NaN        S  
4  22.0      1      1  3101298  12.2875   NaN        S

### Columnas:

PassengerId = Id del pasajero.  
Pclass = Clase del boleto (1 = 1st, 2 = 2nd, 3 = 3rd).  
Name = Nombre del pasajero.  
Sex = Sexo del pasajero.  
Age = Edad del pasajero.  
SibSP = Número de hermanos o cónyuge del pasajero a bordo.  
Parch = Número de padres o hijos del pasajero a bordo.  
Ticket = Número del boleto.  
Fare = Tarifa del pasajero.  
Cabin = Número de cabina del pasajero.  
Embarked = Puerto de embarque (C = Cherburgo, Q = Queenstown, S = Southampton).  

El dataset de entrenamiento contiene una columna adicional en comparación con el dataset de prueba: la columna 'Survived', que indica si el pasajero sobrevivió (1) o no sobrevivió (0). Esta columna va a ser nuestro target para entrenar los modelos.

In [5]:
print(f'Dimensiones del dataset de entrenamiento: {train_data.shape[0]} filas, {train_data.shape[1]} columnas')

Dimensiones del dataset de entrenamiento: 891 filas, 12 columnas


In [6]:
print(f'Dimensiones del dataset de prueba: {test_data.shape[0]} filas, {test_data.shape[1]} columnas')

Dimensiones del dataset de prueba: 418 filas, 11 columnas


Como podemos ver, el dataset de prueba tiene una columna menos y poco menos de la mitad de registros que el dataset de entrenamiento.

---

In [7]:
# Revisamos la información del dataset de entrenamiento.
print('Info datos de entrenamiento: \n')
train_data.info()

Info datos de entrenamiento: 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [8]:
# Revisamos los valores faltantes en el conjunto de entrenamiento.
print('Valores nulos en el dataset de entrenamiento:')
train_data.isnull().sum()

Valores nulos en el dataset de entrenamiento:


PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

Sabemos que en el dataset de entrenamiento tenemos 891 registros, y la información muestra que las siguientes columnas contienen valores nulos:

'Age' = 177 valores nulos.  
'Cabin' = 687 valores nulos.  
'Embarked' = 2 valores nulos.  

Estos valores nulos van a ser trabajados en el archivo 'valores_nulos.ipynb'.

También, podemos ver que tenemos 5 columnas tipo int64, 2 columnas de tipo float64 y 5 columnas de tipo object.

Las columnas de tipo object pueden necesitar transformarse a variables numéricas para ser usadas en los modelos de machine learning.

In [9]:
# Revisamos la información del dataset de entrenamiento.
print('Info datos de testeo: \n')
test_data.info()

Info datos de testeo: 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  418 non-null    int64  
 1   Pclass       418 non-null    int64  
 2   Name         418 non-null    object 
 3   Sex          418 non-null    object 
 4   Age          332 non-null    float64
 5   SibSp        418 non-null    int64  
 6   Parch        418 non-null    int64  
 7   Ticket       418 non-null    object 
 8   Fare         417 non-null    float64
 9   Cabin        91 non-null     object 
 10  Embarked     418 non-null    object 
dtypes: float64(2), int64(4), object(5)
memory usage: 36.1+ KB


In [10]:
# Revisamos los valores faltantes en el conjunto de prueba.
print('Valores nulos en el dataset de prueba:')
test_data.isnull().sum()

Valores nulos en el dataset de prueba:


PassengerId      0
Pclass           0
Name             0
Sex              0
Age             86
SibSp            0
Parch            0
Ticket           0
Fare             1
Cabin          327
Embarked         0
dtype: int64

Por otro lado, en el dataset de prueba, tenemos un total de 418 registros y las siguientes columnas contienen valores nulos:

'Age' = 86 valores nulos.  
'Fare' = 1 valores nulos.  
'Cabin' = 327 valores nulos.  

Estos valores nulos también van a ser trabajados en el archivo 'valores_nulos.ipynb'.

Tenemos 4 columnas tipo int64, 2 columnas de tipo float64 y 5 columnas de tipo object.

De la misma forma que en el dataset de entrenamiento, las columnas de tipo object también van a ser transformadas para entrenar los modelos de machine learning.

---

In [11]:
# Vemos la estadísticas descriptivas de las variables numéricas del dataset de entrenamiento.
print('Estadísticas descriptivas (Dataset Entrenamiento):')
train_data.describe()

Estadísticas descriptivas (Dataset Entrenamiento):


PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

### Count:
Como ya vimos antes, la columna númerica con datos nulos es la de 'Age' contando con un total de 177 valores nulos en los registros.

### Mean (media):
**Survived**: Vemos que aproximadamente el 38.3% de los pasajeros dentro del dataset de entrenamiento sobrevivieron.  
**Pclass**: La media indica que la mayoría de pasajeros estaban en la clase 2 o 3.  
**Age**: La edad promedio de los pasajeros era de 29.7 años.  
**SibSp** y **Parch**: EL número promedio de hermanos/cónyuge y padres/hijos es bastante bajo (0.52 y 0.38), esto nos dice que la mayoría de los pasajeros viajaban solos o con una persona.
**Fare**: El promedio es 32.20, lo que puede indicarnos una fuerte variabilidad en los precios pagados por los boletos.

### Std (desviación estándar):
**Survived**: Tiene una desviación estándar alta (0.49), lo que nos indica una distribución uniforme entre los pasajeros que sobrevivieron y los que no.  
**Fare**: Tiene una desviación estándar alta (50), indicando que algunos pasajeros pagaron voletos muy caros.

### Min (valor mínimo):
Con el valor mínimo podemos ver, por ejemplo, que algunos pasajeros pagaron 0 por sus boletos (Fare), podríamos sacar conclusiones como boletos de cortesía, regalo o alguna situación especial. También vemos que el mínimo de edad es menos de 0 años, ahora sabemos que habían bebés recién nacidos en el Titanic.

### Percentiles (25%, 50%, 75%):

**Pclass**: El 50% de los pasajeros pertenecian a la clase 3.
**Age**: El 25% de los pasajeros tienen menos de 20.12 años, el 50% tienen menos 28 años y el 75% tienen menos de 38 años.
**Fare**: El 25% de los pasajeros pagaron menos de 7.91, el 50% pagó 14.45 o menos y el 75% pagó 31 o menos, aunque los valores de tarifa más altos nos indican una gran variedad.

### Max (valor máximo):

El valor máximo muestra que el pasajero más viejo tenía 80 años y el boleto más caro costó 512.32, esto es bastante caro en comparación con la media.

In [12]:
# Vemos las estadísticas descriptivas de las variables númericas del dataset de prueba.
print('Estadísticas descriptivas (Dataset Prueba):')
test_data.describe()

Estadísticas descriptivas (Dataset Prueba):


PassengerId      Pclass         Age       SibSp       Parch        Fare
count   418.000000  418.000000  332.000000  418.000000  418.000000  417.000000
mean   1100.500000    2.265550   30.272590    0.447368    0.392344   35.627188
std     120.810458    0.841838   14.181209    0.896760    0.981429   55.907576
min     892.000000    1.000000    0.170000    0.000000    0.000000    0.000000
25%     996.250000    1.000000   21.000000    0.000000    0.000000    7.895800
50%    1100.500000    3.000000   27.000000    0.000000    0.000000   14.454200
75%    1204.750000    3.000000   39.000000    1.000000    0.000000   31.500000
max    1309.000000    3.000000   76.000000    8.000000    9.000000  512.329200

### Count:
La columna 'Age' tiene 86 valores nulos y la columna 'Fare' tiene un solo valor nulo.

### Mean (media):
**Pclass**: La media indica que la mayoría de pasajeros estaban en la clase 2 o 3, como en el dataset de entrenamiento.  
**Age**: La edad promedio de los pasajeros era de 30.27 años, un poco más alta que en el dataset de entrenamiento.  
**Fare**: El promedio del boleto es 35.62, un poco más alto que en el dataset de entrenamiento.  

### Std (desviación estándar):
**Fare**: Tiene una desviación estándar alta (55). Como en el dataset de entrenamiento, también vemos que algunos pasajeros pagaron boletos más caros que otros.

### Min (valor mínimo):
También podemos ver que, en el dataset de prueba, hay pasajeros que pagaron 0 por sus boletos.  

### Percentiles (25%, 50%, 75%):

**Age**: El 25% de los pasajeros tienen menos de 21 años, el 50% tiene menos de 27 años y el 75% tiene menos de 39 años.
**Fare**: La mayoría de los pasajeros pagó tarifas iguales a las del dataset de entrenamiento, con el 75% pagando hasta 31.50.

### Max (valor máximo):

La tarifa más alta pagada por un boleto es la misma en ambos datasets, y la edad máxoma es de 76 años en el dataset de prueba, un poco más baja que en el dataset de entrenamiento.

---